geopandas/tests/test_cmapplotting.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import absolute_import, division
+
+import itertools
+import warnings
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+
+from shapely.geometry import Point
+
+from geopandas import GeoSeries, GeoDataFrame, plotting
+from geopandas.tests.test_plotting import _check_colors
+
+import pytest
+
+@pytest.fixture(autouse=True)
+def close_figures(request):
+    yield
+    plt.close('all')
+
+class TestCMapPlotting:
+    def setup_method(self):
+        # Intializing the dataframe
+        self.N = 10
+        self.points = GeoSeries(Point(i, i) for i in range(self.N)).buffer(0.5)
+        values = np.arange(self.N)
+        levels = ['1','1','2','4','5','1','5','4','3','2']
+        self.df = GeoDataFrame({'geometry': self.points, 'values': values, 'levels': levels})
+
+    def test_customnorm(self):
+        # test color without additional parameters
+        ax = self.df.plot(column='values')
+        actual_color = ax.collections[0].get_facecolors()
+        cmap = ax.collections[0].get_cmap()
+        expected_color = cmap(np.arange(self.N) / (self.N - 1))
+        np.testing.assert_array_equal(actual_color, expected_color)
+
+        # test color with norm
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,3,6,8], cmap.N)
+        ax = self.df.plot(column='values', norm = norm)
+        actual_color = ax.collections[0].get_facecolors()
+        cmap = ax.collections[0].get_cmap()
+        expected_color = cmap(norm(list(self.df['values'])))
+        np.testing.assert_array_equal(actual_color, expected_color)
+
+    def test_minmax(self):
+        # test colorbar min,max without additional parameters
+        ax = self.df.plot(column='values', legend=True)
+        colorbar_color = ax.get_figure().axes[1].collections[0].get_clim()
+        point_color = ax.collections[0].get_clim()
+        assert colorbar_color == point_color
+
+        # test colorbar min,max with norm
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,3,6,8], cmap.N)
+        ax = self.df.plot(column='values', legend=True, norm = norm)
+        colorbar_color = ax.get_figure().axes[1].collections[0].get_clim()
+        point_color = ax.collections[0].get_clim()
+        assert colorbar_color == point_color
```

Comment:
Can you assert here also that those are equal to the expected value (which is (1, 8) I think) ?

Reply:
Can you update for this comment?

Reply:
I'll put a docstring to notify the values of both min and max values as (1,8) and put the respective colors in the docstring.

Reply:
I would actually test this instead of putting it in the docstring